src/rust/generators/sysmon-generator/src/api.rs

```diff
@@ -22,7 +22,9 @@ impl GeneratorApi for SysmonGenerator {
         &self,
         request: RunGeneratorRequest,
     ) -> Result<RunGeneratorResponse, Self::Error> {
-        let sysmon_event = SysmonEvent::from_str(std::str::from_utf8(&request.data)?)?;
+        let input_utf8 = std::str::from_utf8(&request.data)?;
+        let events: Vec<_> = sysmon_parser::parse_events(input_utf8).collect();
+        let sysmon_event: SysmonEvent = expect_one_event(events)?;
```

Comment:
This is not a limitation of the generator - it can produce a graph result from many events just fine.

Enforcing one event here is purely artificial? Seems better served elsewhere.

Reply:
Looks like this has indeed been changed to only process a single event (https://github.com/grapl-security/grapl/blob/main/src/rust/generators/sysmon-generator/src/api.rs#L27),

but IMO we should revert the change that made it only handle one at a time, and if we want to enforce that we should do it elsewhere and keep this as flexible as possible.

Reply:
I think where we ended up in our discussion today was that the pipeline (e.g. everything after pipeline-ingress) operates on single events. So in the absence of a technical requirement to support processing more than one event per `RawLog`, and with the clear expectation that clients won't send more than one event per `RawLog`, it's hard to see the value in maintaining code which _expects_ more than one event per `RawLog`... that seems like it'll suffer bitrot.

See https://grapl-internal.slack.com/archives/C03JNJ57A9G/p1659457692934169

Reply:
The previous version did not expect more than one event, it was flexible in being able to handle one or more events. If the backend wanted to send a single event as a time, that was fine, the code worked just the same and there was no additional maintenance needed for that flexibility, and there were no bits rotting.

If anything, this PR is additional maintenance, one to help enforce a requirement of a different system, which IMO it shouldn't do.

Reply:
It looks like the discussion you linked to runs too deep for this PR tho, I'll remove my request for changes.

Reply:
Yeah I think it's fine if this code can handle more than one event, we just need to make it sure with tests that exercise both the one event and the many events case.

Reply:
> In that case, a parser which understands more than one event at a time might make sense?

The parser and generator operated on multiple events, which it got from `sysmon_parser::parse_events`, just as it would now in this PR. Previously: 
https://github.com/grapl-security/grapl/blob/2e1558f47fc01cb5e249dd46624a4686e7d928ff/src/rust/generators/sysmon-generator/src/serialization.rs#L16

where the generator folded multiple `GraphDescription`s into a single: 
https://github.com/grapl-security/grapl/blob/2e1558f47fc01cb5e249dd46624a4686e7d928ff/src/rust/generators/sysmon-generator/src/generator.rs#L86

Reply:
> Yeah I think it's fine if this code can handle more than one event, we just need to make it sure with tests that exercise both the one event and the many events case.
> 
> And, to be clear, the original change which surfaced this bug was my mistake 😞. I was in a hurry to get a generator running in the Kafka pipeline and made a bad call.

All good, no worries - we can revisit this when we're ready, but .. I don't think today is that day. 👍